CHANGELOG.md

```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Remove random function from ink!.
```

Comment:
```suggestion
### Removed
- Remove random function ‒ [#1442](https://github.com/paritytech/ink/pull/1442)
```

Reply:
Could you please add some more information in a section `### How to replace random` here?

My concern is that we're just removing it without providing guidance on how to substitute this functionality. 

Is there no pallet out there which we could create a chain extension with?

Reply:
On ethereum contracts need to look for randomness on their own anyways. They rely on oracle contracts AFAIK. If we had some reliable way to provide randomness in the general case we wouldn't remove it. I don't think there is anything at the moment.

@burdges  has some ideas but nothing readily available. So we can't really give any advice beyond "good luck bro".

Reply:
This pseudo random function was useful enough for our contract but now its removed. Any guidance on how to recreate this functionality would be appreciated.

Reply:
We need a random seed per block that is _good enough_ to randomly read a value from a vector within our contract. The operation is not high value and the seed calculated by the validators was fulfilling this purpose very well. Are the validators no longer committing this random seed to each block?

Could the current block hash be made available in `ink!` as a substitute?

Reply:
> We need a random seed per block that is _good enough_ to randomly _read_ a value from a vector within our contract. There is no transaction involved. The operation is not high value and the seed calculated by the validators was fulfilling this purpose very well. Are the validators no longer committing this random seed to each block?
> 
> Could the current block hash be made available in `ink!` as a substitute?

We would need **way** for information to give some advice and to determine if using predictable randomness is actually safe.  I recommend opening an issue on [stack exchange](https://substrate.stackexchange.com).

Reply:
I've provided more info on [stack exchange](https://substrate.stackexchange.com/questions/6745/how-can-i-generate-a-random-seed-now-that-self-env-random-is-removed-in-ink).